core/state/accessors.go

```diff
@@ -0,0 +1,133 @@
+package state
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/db"
+)
+
+func GetStateObject(r db.KeyValueReader, state *State, addr *felt.Felt) (*stateObject, error) {
+	contract, err := GetContract(r, addr)
+	if err != nil {
+		return nil, err
+	}
+
+	return newStateObject(state, addr, &contract), nil
+}
+
+// Wrapper around getContract which checks if a contract is deployed
```

Comment:
I think we can improve this comment, I understand what you meant but it is not very clear. 

One suggestion:
```suggestion
// Get a contract instance from the database. If it doesn't exist returns a contract not deployed error
```